### PR TITLE
Added Lua annotation plugin documentation

### PR DESCRIPTION
**Pull request by adrianherrera**
Plus some other minor fixes based on feedback I've received from various people.

### CODE REVIEW

File: src/Tools/ForkProfiler.rst [hunk 1]
```diff
@@ -5,7 +5,8 @@ Fork Profiler
 The fork profiler tool lists all the program counters that caused a fork. This allows us to quickly identify the
 program's hot spots that cause path explosion problems (as discussed in the `FAQ <../FAQ.rst>`_).
 
-NOTE: Function names will be available only if the analysed program and libraries are built with debug symbols support.
+**NOTE**: Function names will be available only if the analysed program and libraries are built with debug symbols
```
Comment (vitalych) [inline src/Tools/ForkProfiler.rst]: I think we use american spelling (s=>z) :)

File: src/s2e-env.rst [hunk 2]
```diff
@@ -52,7 +52,9 @@ To create a new S2E environment in ``/home/user/s2e``, run:
 
 This will fetch the required source code, install S2E's dependencies (via apt) and create the directory structure
 described `here <https://github.com/s2e/s2e-env/blob/master/README.md>`_. If you want to skip the dependency
-installation step (e.g. if you have already installed the dependencies) use the ``--skip-dependencies`` flag.
+installation step (e.g. if you have already installed the dependencies) use the ``--skip-dependencies`` flag. Note that
+if you are using Ubuntu 14.04, you must install CMake manually - S2E requires version 3.4.3 or newer, which is not
+available in the Ubuntu 14.04 repositories.
```
Comment (vitalych) [inline src/s2e-env.rst]: Perhaps add a check to s2e-env?

File: src/s2e-env.rst [hunk 3]
```diff
@@ -119,6 +121,9 @@ in the following repositories:
 * `guest-images <https://github.com/S2E/guest-images>`_
 * `s2e-linux-kernel <https://github.com/S2E/s2e-linux-kernel>`_
 
+**NOTE**: The image build process caches intermediate build output in `.tmp-output` that can grow quite large. Once
+the images have been built you may wish to delete this directory if disk space is an issue
```
Comment (vitalych) [inline src/s2e-env.rst]: Period at end of sentence.

File: src/Plugins/FunctionMonitor.rst [hunk 4]
```diff
@@ -43,9 +43,7 @@ For example, to monitor the kernel-mode function located at ``0xC00F012``, speci
                                           S2EExecutionState *state,
                                           TranslationBlock *tb,
                                           uint64_t pc) {
-        FunctionMonitor::CallSignal *callSignal;
-
-        // 4. Obtain the address of the function to be monitored
+        FunctionMonitor::CallSignal *callSignal        // 4. Obtain the address of the function to be monitored
```
Comment (vitalych) [inline src/Plugins/FunctionMonitor.rst]: I wouldn't put comment on the same line

File: src/Tools/ForkProfiler.rst (hunk 1 above)
Comment (adrianherrera) [reply src/Tools/ForkProfiler.rst]: I've been back in Australia for too long it seems :)

File: src/s2e-env.rst (hunk 2 above)
Comment (adrianherrera) [reply src/s2e-env.rst]: Good idea! I'll add the check.

**Comment by adrianherrera** [comment]
Add fixed.

File: src/Plugins/FunctionMonitor.rst (hunk 4 above)
Comment (adrianherrera) [reply src/Plugins/FunctionMonitor.rst]: I'm not sure how that happened, it's not even valid code (it's missing a semicolon) :\

**Pull request merged**